pkg/addon/addon.go

```diff
@@ -581,7 +592,21 @@ func RenderApp(ctx context.Context, addon *InstallPackage, config *rest.Config,
 		app.Spec.Components = append(app.Spec.Components, *comp)
 	}
 
+	deployClusters := getClusters(args)
+
 	switch {
+	case len(deployClusters) > 0:
+		if app.Spec.Policies == nil {
+			app.Spec.Policies = []v1beta1.AppPolicy{}
+		}
+		body, _ := json.Marshal(map[string][]string{types.ClustersArg: deployClusters})
+		app.Spec.Policies = append(app.Spec.Policies, v1beta1.AppPolicy{
+			Name:       "specified-addon-clusters",
+			Type:       v1alpha1.TopologyPolicyType,
+			Properties: &runtime.RawExtension{Raw: body},
+		})
+		// if users have already specified workflow in addon, this won't work
```

Comment:
IMO these lines should be put in case  `isDeployToRuntimeOnly(addon)`.  Only when an addon's `deployTo.runtimeCluster`  field is true, user can choose cluster.

Reply:
When users specify the clusters argument, we can ignore this configuration. We should follow the users input on deploy time.

Reply:
followed your advice